### PR TITLE
Usability improvements

### PR DESCRIPTION
**Pull request by BRHoward**
- Instead of replacing the conversation with each 'get' new messages are spliced into client side array
- New messages show a toast notification
- Can search through user list
- Conversations with more recent changes are shown at the top
- Users can clear conversations or show all messages in a conversation

### CODE REVIEW

**Comment by BRHoward** [comment]
@stevenhillcox @jebbinSCL @sievins

File: public/js/main.js [hunk 1]
```diff
@@ -78,11 +80,17 @@
                 });
         }
 
-        function getConversations() {
+        function getConversations(firstLoad) {
```
Comment (jebbinSCL) [inline public/js/main.js]: is this getConversations, or refreshConversations? It is doing more than a simple get.

**Comment by BRHoward** [comment]
@jebbinSCL

File: public/js/main.js [hunk 2]
```diff
@@ -113,13 +121,57 @@
             });
         }
 
+        function displayMessageNotification(message) {
+            var messageFrom = getUserFromId(message.userid).name;
+            toastr.info(message.text, "Message from " + messageFrom);
+        }
+
+        function setIfConversationIsCleared(conversationId, cleared) {
```
Comment (jebbinSCL) [inline public/js/main.js]: Rename to setClearedForConversationMessages, or similar. You might be able to think of a better name than that?

**Comment by jebbinSCL** [comment]
:ship:

**Pull request merged**